### PR TITLE
Add static code analysis

### PR DESCRIPTION
**Pull request by Renny4Real**
## Link to JIRA ticket

https://hackney.atlassian.net/browse/PA-382

## Describe this PR

Adds fixes to errors raised by static code analysis 

### *What changes have we introduced*

Added a separate `.editorconfig` file for the test project which would only allow tests to have underscores in their indentifiers.
The test project will still inherit configuration from the top-level `.editorconfig` file.

#### _Checklist_

- [ ] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [ ] Added tests to cover all new production code
- [ ] Added comments to the README or updated relevant documentation _(add link to documentation)_, where necessary.
- [ ] Checked all code for possible refactoring
- [X] Code pipeline builds correctly

### CODE REVIEW

File: AddressesAPI/V1/Domain/Address.cs [hunk 1]
```diff
@@ -2,13 +2,13 @@ namespace AddressesAPI.V1.Domain
...
     {
-        public string Line1 { get; set; }
-        public string Line2 { get; set; }
-        public string Line3 { get; set; }
-        public string Line4 { get; set; }
-        public string Town { get; set; }
-        public string Postcode { get; set; }
-        public long UPRN { get; set; }
+        public new string Line1 { get; set; }
+        public new string Line2 { get; set; }
```
Comment (elcorbs) [inline AddressesAPI/V1/Domain/Address.cs:6]: What rule is this for? Maybe it can be ignored..

File: AddressesAPI/V1/Gateways/AddressesGateway.cs [hunk 2]
```diff
@@ -2,6 +2,9 @@
 using System.Threading.Tasks;
 using AddressesAPI.V1.Domain;
 using AddressCrossReference = AddressesAPI.V1.Domain.AddressCrossReference;
+#pragma warning disable 1998
```
Comment (elcorbs) [inline AddressesAPI/V1/Gateways/AddressesGateway.cs:5]: This is cool

File: AddressesAPI/V1/Domain/Address.cs (hunk 1 above)
Comment (humulla) [reply AddressesAPI/V1/Domain/Address.cs:6]: I don't understand this change too

**Review by humulla — APPROVED** [review]
Looks good, just an clarification needed for domain Address change

File: AddressesAPI/V1/Domain/Address.cs (hunk 1 above)
Comment (Renny4Real) [reply AddressesAPI/V1/Domain/Address.cs:6]: It was inheriting from a class that had the parameters named the same. It was a `CS0108` [compiler warning](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/cs0108)

Comment (elcorbs) [reply AddressesAPI/V1/Domain/Address.cs:6]: I think the lines can all just be removed then

Comment (Renny4Real) [reply AddressesAPI/V1/Domain/Address.cs:6]: The suggestion was to add the `new` keyword. I wasn't sure why the inheritance was there but didn't want to remove it as part of adding static code analysis.

Comment (Renny4Real) [reply AddressesAPI/V1/Domain/Address.cs:6]: But I can remove them 👍

Comment (elcorbs) [reply AddressesAPI/V1/Domain/Address.cs:6]: I added the inheritance so there was a relationship between the two address types that were there previously and just forgot that they could be removed if they are in the parent class. Thanks!

**Pull request merged**